### PR TITLE
Creation of userlist service

### PR DESCRIPTION
**Pull request by reidbaker**
This moves all our code to the new userlist service. 
A side effect was that I made the musiclibarary fragment use a service locator for the music library service.

There is also code to null check some buttons that were crashing on rotation.

### CODE REVIEW

File: SoundStream/src/com/lastcrusade/soundstream/components/MenuFragment.java [hunk 1]
```diff
@@ -130,4 +147,25 @@ public void onReceiveAction(Context context, Intent intent) {
     private void unregisterReceivers() {
         this.registrar.unregister();
     }
+
+    private UserList getUserListFromService(){
+        UserList activeUsers = new UserList();
```
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/components/MenuFragment.java]: It seems like we make this variable for no real reason. We could just return a new UserList in the bottom return

File: SoundStream/src/com/lastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 2]
```diff
@@ -44,81 +45,72 @@
...
-            boundToService = false;
-        }
-    };
-
-
     @Override
     public void onCreate(Bundle savedInstanceState){
         super.onCreate(savedInstanceState);
 
+        //make a new music list adapter and give it an empty list of songs and empty list of users to use until the service is connected
```
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/components/MusicLibraryFragment.java]: split this onto two lines?

File: SoundStream/src/com/lastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 3]
```diff
@@ -170,6 +161,49 @@ protected PlaylistService getPlaylistService() {
...
+        if(musicLibraryService != null){
+            library = new ArrayList<SongMetadata>(musicLibraryService.getLibrary());
+        } else {
+            Log.i(TAG, "MusicLibarysService null, returning empty library");
+        }
+        return library;
+    }
+
+
+    private UserList getUserListFromService(){
```
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/components/MusicLibraryFragment.java:200]: Same as before, though you could also change it to do what the music library version does

File: SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java [hunk 4]
```diff
@@ -82,6 +86,18 @@ public void onServiceBound() {
             }
         });
 
+        NetworkFragment.this.adapter = new UserListAdapter(getActivity(), new UserList(), false );
```
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java]: Is there a reason that you include NetworkFragment before this.adapter?

File: SoundStream/src/com/lastcrusade/soundstream/components/ConnectFragment.java [hunk 5]
```diff
@@ -156,19 +156,21 @@ public void onReceiveAction(Context context, Intent intent) {
...
-                    case BluetoothAdapter.SCAN_MODE_CONNECTABLE:
-                        connectButton.setEnabled(true);
-                        break;
-                    case BluetoothAdapter.SCAN_MODE_CONNECTABLE_DISCOVERABLE:
-                        connectButton.setEnabled(false);
-                        break;
-                    default:
-                        Log.wtf(TAG, "Recieved scan mode changed with unknown mode");
-                        break;
+                    if(connectButton != null){
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/components/ConnectFragment.java:159]: Maybe not part of this pull request, but we should make sure we're not "paused"...that way, we don't respond to these actions when performed outside the app.
